Source/JavaScriptCore/parser/ASTBuilder.h

```diff
@@ -1106,8 +1111,17 @@ class ASTBuilder {
         }
         int m_features;
         int m_numConstants;
+        bool m_usesArgumentsLength { false };
     };
 
+    void checkArgumentsLengthModification(const ExpressionNode* node)
+    {
+        // Since we exclude pattern `arguments.length` to enable ArgumentsFeature,
+        // we need re-enable ArgumentsFeature for `arguments.length` modification.
+        if (node && node->isArgumentsLengthAccess(m_vm))
+            usesArguments();
+    }
```

Comment:
I don't think this is the right approach. We should disable `arguments.length` if we found `arguments` identifier resolves regardless of `arguments.xxx = ` assignment. If it has a getter, then it can do whatever we want.

Reply:
We need this for these cases:

```
function argumentsLengthAssignment() {
    arguments.length = 3;
    shouldBe(arguments.length, 3);
}
argumentsLengthAssignment(1);
```
and
```
function argumentsLengthIncAndDec() {
    arguments.length++;
    arguments.length--;
    ++arguments.length;
    --arguments.length;
    shouldBe(arguments.length, 0);
}
argumentsLengthIncAndDec();
```